.github/workflows/ci.yml

```diff
@@ -107,18 +107,35 @@ jobs:
     strategy:
       matrix:
         go-version: ["~1.22.0", "~1.21.3"]
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        # GitHub Actions does not support arm* architectures on default
-        # runners. It is possible to accomplish this with a self-hosted runner
-        # if we want to add this in the future:
-        # https://docs.github.com/en/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow
-        arch: ["386", amd64]
-        exclude:
-          # Not a supported Go OS/architecture.
+        platform:
+          - os: ubuntu-latest
+            arch: "386"
+          - os: ubuntu-latest
+            arch: amd64
           - os: macos-latest
+            arch: amd64
+          - os: windows-latest
             arch: "386"
-    runs-on: ${{ matrix.os }}
+          - os: windows-latest
+            arch: amd64
+          - os: actuated-arm64-1cpu-1gb
+            arch: arm64
+    runs-on: ${{ matrix.platform.os }}
     steps:
+      # Setup VMMeter for actuated jobs, so we can track CPU/memory usage and
+      # finetune our instance size accordingly.
+      # See https://actuated.dev/blog/arm-ci-cncf-ampere
+      - name: Set up arkade
+        uses: alexellis/setup-arkade@v3
+        if: ${{ startsWith(matrix.platform.os, 'actuated') }}
+      - name: Install vmmeter
+        if: ${{ startsWith(matrix.platform.os, 'actuated') }}
+        run: |
+          sudo -E arkade oci install ghcr.io/openfaasltd/vmmeter:latest --path /usr/local/bin/
+      - name: Run vmmeter
+        uses: self-actuated/vmmeter-action@v1
+        if: ${{ startsWith(matrix.platform.os, 'actuated') }}
```

Comment:
Do we really need this? I suggest removing.

```suggestion
```

Reply:
We'd need it if there are stability issues due to the number of CPUs/amount of RAM.
I kept it because it shouldn't impact anything else, but I can remove it too.

Reply:
I think this should be added to a troubleshooting handbook instead of running each time.

Reply:
It's in https://github.com/open-telemetry/community/pull/1983